### PR TITLE
feat(components): add additional config for graph output command

### PR DESCRIPTION
**Pull request by esensar**
This adds additional `graph` config to all components which enables adding any `node_attributes` to the generated graph.

Closes: #21043

### CODE REVIEW

**Comment by esensar** [comment]
This currently only supports string values for each attribute. It still seems to work fine with `dot`, but it may be a bit weird to use in configuration.

**Review by jszwedko — COMMENTED** [review]
Nice! This seems useful. Thanks @esensar . I left a few comments in-line, and there are some clippy failures, but this looks pretty good otherwise. I'll try to play around with it locally.

**Comment by esensar** [comment]
Does it make sense to keep these attributes as a String -> String map? It might be weird for some attributes that are of other types, even though it will still work fine.

Besides that, I think it is fairly simple to add `edge_attributes` (which would apply only to transforms and sinks) and global attributes which would apply to the whole graph. I just want to check if that is alright before adding.

File: changelog.d/21194_dot_graph_attributes.feature.md
Comment (neko-dd) [inline changelog.d/21194_dot_graph_attributes.feature.md:1]:
```suggestion
Adds support for additional graph configuration on each component so that users can add arbitrary graphviz node attributes when generating a graph via `vector graph`.
```

File: website/cue/reference/components/base/sources.cue
Comment (neko-dd) [inline website/cue/reference/components/base/sources.cue:40]:
```suggestion
			Similar to common proxy configuration convention, you can set different proxies
			to use based on the type of traffic being proxied. You can also set specific hosts that
			should not be proxied.
```

**Review by neko-dd — APPROVED** [review]
Some documentation related suggestions, but otherwise LGTM. Please wait for technical review to also approve before merging. Thanks for the contribution!

File: website/cue/reference/components/base/sources.cue
Comment (esensar) [reply website/cue/reference/components/base/sources.cue:40]: I am not sure how this ended up here. It seems that changes in proxy config were not updated in docs. I will update the proposed changes though, since it already ended up in the PR somehow :smile:

**Comment by jszwedko** [comment]
> Does it make sense to keep these attributes as a String -> String map? It might be weird for some attributes that are of other types, even though it will still work fine.

It seems like strings work fine with graphviz even if the values are actually other types. I'd say we can roll with strings and add support for other types later if users are confused (this would be an additive rather than a breaking change).

> Besides that, I think it is fairly simple to add `edge_attributes` (which would apply only to transforms and sinks) and global attributes which would apply to the whole graph. I just want to check if that is alright before adding.

Edge attributes is a bit tricky since there could be multiple edges to each component. Were you thinking it'd apply the attributes to all incoming edges? One could imagine it applying to outgoing edges. Or maybe we should support both?

Global graph attributes makes sense to me. We could add a top-level `graph` option.

Would you want me to merge this first and you follow up with PRs for the edge and global attributes?

**Comment by esensar** [comment]
> > Does it make sense to keep these attributes as a String -> String map? It might be weird for some attributes that are of other types, even though it will still work fine.
> 
> It seems like strings work fine with graphviz even if the values are actually other types. I'd say we can roll with strings and add support for other types later if users are confused (this would be an additive rather than a breaking change).
> 
> > Besides that, I think it is fairly simple to add `edge_attributes` (which would apply only to transforms and sinks) and global attributes which would apply to the whole graph. I just want to check if that is alright before adding.
> 
> Edge attributes is a bit tricky since there could be multiple edges to each component. Were you thinking it'd apply the attributes to all incoming edges? One could imagine it applying to outgoing edges. Or maybe we should support both?
> 
> Global graph attributes makes sense to me. We could add a top-level `graph` option.
> 
> Would you want me to merge this first and you follow up with PRs for the edge and global attributes?

I think we can do a follow up PR for these. None of these should be breaking, so I think it should be alright.

**Comment by jszwedko** [comment]
Thanks @esensar !

**Pull request merged**
